addons/payment/controllers/portal.py

```diff
@@ -306,7 +304,7 @@ def _create_transaction(
 
     @staticmethod
     def _update_landing_route(tx_sudo):
-        """ Recompute the access token stored in the landing route of validation transactions.
+        """ Add the mandatory parameters to the route and recompute the access_token if needed
 
         The generic landing route require the tx id and access token to be provided, since there is
         no document to rely on. The access token is recomputed in case we are dealing with a
```

Comment:
Let's do as the docstring suggests: recompute the access token if the operation is a validation and use the one from the kwarg otherwise.

```suggestion
        The generic landing route requires the tx id and access token to be provided since there is
        no document to rely on. The access token is recomputed in case we are dealing with a
        validation transaction (acquirer-specific amount and currency).

       :param str access_token: The access token used to authenticate the partner
```

Reply:
Please make sure that the flow of the donation snippet is still valid after that. I believe that it overrides `_update_landing_route` or something like that.